exporters/trace/jaeger/env.go

```diff
@@ -0,0 +1,146 @@
+
+package jaeger
+
+import (
+	"os"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/kv/value"
+)
+
+// Environment variable names
+const (
+	// The service name.
+	envServiceName = "JAEGER_SERVICE_NAME"
+	// Whether the exporter is disabled or not. (default false).
+	envDisabled = "JAEGER_DISABLED"
+	// A comma separated list of name=value tracer-level tags, which get added to all reported spans.
+	// The value can also refer to an environment variable using the format ${envVarName:defaultValue}.
+	envTags = "JAEGER_TAGS"
+	// The HTTP endpoint for sending spans directly to a collector,
+	// i.e. http://jaeger-collector:14268/api/traces.
+	envEndpoint = "JAEGER_ENDPOINT"
+	// Username to send as part of "Basic" authentication to the collector endpoint.
+	envUser = "JAEGER_USER"
+	// Password to send as part of "Basic" authentication to the collector endpoint.
+	envPassword = "JAEGER_PASSWORD"
+)
+
+// CollectorEndpointFromEnv return environment variable value of JAEGER_ENDPOINT
+func CollectorEndpointFromEnv() string {
+	return os.Getenv(envEndpoint)
+}
+
+// WithCollectorEndpointOptionFromEnv uses environment variables to set the username and password
+// if basic auth is required.
+func WithCollectorEndpointOptionFromEnv() CollectorEndpointOption {
+	return func(o *CollectorEndpointOptions) {
+		if e := os.Getenv(envUser); e != "" {
+			o.username = e
+		}
+		if e := os.Getenv(envPassword); e != "" {
+			o.password = os.Getenv(envPassword)
+		}
+	}
+}
+
+// WithDisabledFromEnv uses environment variables and overrides disabled field.
+func WithDisabledFromEnv() Option {
+	return func(o *options) {
+		if e := os.Getenv(envDisabled); e != "" {
+			if v, err := strconv.ParseBool(e); err == nil {
+				o.Disabled = v
+			}
+		}
+	}
+}
+
+// WithProcessFromEnv parse environment variables into jaeger exporter's Process.
+func ProcessFromEnv() Process {
+	var p Process
+	if e := os.Getenv(envServiceName); e != "" {
+		p.ServiceName = e
+	}
+	if e := os.Getenv(envTags); e != "" {
+		p.Tags = parseTags(e)
+	}
+
+	return p
+}
+
+// WithProcessFromEnv uses environment variables and overrides jaeger exporter's Process.
+func WithProcessFromEnv() Option {
+	return func(o *options) {
+		p := ProcessFromEnv()
+		if p.ServiceName != "" {
+			o.Process.ServiceName = p.ServiceName
+		}
+		if len(p.Tags) != 0 {
+			o.Process.Tags = p.Tags
+		}
+	}
+}
+
+// parseTags parses the given string into a collection of Tags.
+// Spec for this value:
+// - comma separated list of key=value
+// - value can be specified using the notation ${envVar:defaultValue}, where `envVar`
+// is an environment variable and `defaultValue` is the value to use in case the env var is not set
+func parseTags(sTags string) []kv.KeyValue {
+	pairs := strings.Split(sTags, ",")
+	tags := make([]kv.KeyValue, len(pairs))
+	for i, p := range pairs {
+		field := strings.SplitN(p, "=", 2)
+		k, v := strings.TrimSpace(field[0]), strings.TrimSpace(field[1])
```

Comment:
This will panic if there is a malformed tag (e.g. `a=1,b,c=3`). I don't think we want to panic here, maybe we report using the `global.Handler`?

We should also specify how malformed tags are handled in the doc string.